### ISSUE TITLE
Add `[[warn_unused_result]]` for ConnectionHandle

### ISSUE THREAD

**Issue by LeonMatthesKDAB** [KDAB/KDBindings#43]
Forgetting to disconnect a connection can lead to serious issues.

So we should allow users to at least get a warning in that case.

However, we likely want to put this behind a flag of some sort to not break existing code.

**Comment by lemirep** [comment]
This is nice for typical use cases where you'd need to keep the handle because of lifetime considerations.
However, when dealing with Property and their bundled signals,  we now get spawned with warnings when doing the simplest of valueChanged connect. How could we go about having the warning except for signals from Properties?

```
struct SomeStruct {
    KDBindings::Property<float> myProp{0.0f};
    float m_value{0.0f};

    SomeStruct() {
          // Perfectly safe to ignore the handle in this case
          myProp.valueChanged().connect([this] {
              m_value = myProp();
          });
    }
```

**Comment by LeonMatthesKDAB** [comment]
What we have in CXX-Qt is a [`release`](https://docs.rs/cxx-qt-lib/0.6.1/cxx_qt_lib/struct.QMetaObjectConnection.html#method.release) method on our equivalent of a ConnectionHandle.

Maybe we can create a similar method to silence the warning :thinking: 
Then the code would become:
```
struct SomeStruct {
    KDBindings::Property<float> myProp{0.0f};
    float m_value{0.0f};

    SomeStruct() {
          myProp.valueChanged().connect([this] {
              m_value = myProp();
          }).release(); // <-- specifies that this ConnectionHandle is safe to ignore
    }
```

I'm not entirely sure we can make this work, but would be willing to try if you think it's an acceptable solution.

While it still makes you write a bit more code than before, it makes the intention a lot clearer.
Basically: It shouldn't be possible to implicitly create a lifetime/memory safety issue.
With this design, you at least need to be clear about it.

I'm also open to suggestions on the name of the method. Maybe `release` isn't quite the right fit for KDBindings :thinking: 
Could also be `ignore`, `unmanaged`, or something similar.

**Comment by lemirep** [comment]
That sounds good. I'm fine with release since it maps to similar behavior in unique_ptr

**Comment by MiKom** [comment]
What I do in KDUtils right now is:
```
#include <tuple>
//...
std::ignore = prop.connect([this]{ ... });
```
